### PR TITLE
reboot-util: several cleanups

### PR DESCRIPTION
**Pull request by yuwata**
Follow-ups for #25226.

### CODE REVIEW

File: src/shared/reboot-util.c [hunk 1]
```diff
@@ -1,11 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include <errno.h>
-#include <error.h>
-#include <fcntl.h>
```
Comment (sthibaul) [inline src/shared/reboot-util.c:5]: ? `fcntl.h` is still needed for `open`

Comment (yuwata) [reply src/shared/reboot-util.c:5]: Yeah, but "fileio.h" includes it.

File: src/shared/meson.build [hunk 2]
```diff
@@ -465,8 +465,8 @@ libshared_deps = [threads,
                   libseccomp,
                   libselinux,
                   libzstd,
-                  libxz,
-                  libxenctrl]
+                  libxenctrl,
+                  libxz]
```
Comment (poettering) [inline src/shared/meson.build]: hmm? this does not look right. libzstd should be last

Comment (yuwata) [reply src/shared/meson.build]: Indeed. Fixed.

**Review by bluca — APPROVED** [review]

**Pull request merged**